projects/plugins/jetpack/extensions/shared/components/product-management-controls/index.js

```diff
@@ -0,0 +1,67 @@
+/**
+ * WordPress dependencies
+ */
+import { BlockControls } from '@wordpress/block-editor';
+import { useSelect } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import { API_STATE_CONNECTED } from './constants';
+import ProductManagementInspectorControl from './inspector-control';
+import { jetpackMembershipProductsStore } from './store';
+import ProductManagementToolbarControl from './toolbar-control';
+import InvalidProductWarning from './invalid-product-warning';
+import StripeConnectToolbarButton from '../stripe-connect-toolbar-button';
+
+import './style.scss';
+
+export default function ProductManagementControls( {
+	allowCreateOneTimeInterval = true,
+	selectedProductId = 0,
+	setSelectedProductId = () => {},
+} ) {
+	const products = useSelect(
+		select =>
+			select( jetpackMembershipProductsStore ).getProducts(
+				selectedProductId,
+				setSelectedProductId
+			),
+		[]
+	);
```

Comment:
This selector only fetches data on mount, which is the same behaviour as before.